### DISCUSSION TITLE
[OPINION] The Live Preview feature should be easily turned off.

### DISCUSSION THREAD

**Discussion by MerryPanda** [phcode-dev/phoenix#1534] [General]
I don't need Live Preview but can't find the way to turn it off. Please add an option.

**Comment by abose** [comment]
![image](https://github.com/phcode-dev/phoenix/assets/5336369/1ba0fe21-5c8e-4d1a-be54-fec7fa945d0d)
Please click on the lightning button to turn off live preview.

**Reply by MerryPanda** [reply]
Thanks, but it is On on start always

**Reply by abose** [reply]
Ah! In desktop builds from next release, we have made a change so that live preview only comes up on startup if its an html file that is open in editor. Else it wont come up at startup.will that help?

**Comment by MerryPanda** [comment]
> Ah! In desktop builds from next release, we have made a change so that live preview only comes up on startup if its an html file that is open in editor. Else it wont come up at startup.will that help?

I would rather stay it turned off by default. If somebody needs it, they could just turn it on. Why to enforce it?

**Reply by abose** [reply]
Let me see if i can streamline the first launch popup situation for new users. But you will not see these popups after first launch.

**Comment by abose** [comment]
This is now possible via live preview settings.

**Comment by MerryPanda** [comment]
> This is now possible via live preview settings.

Yeah, thanks. But when I open a new project there is still a pop up that suggests to edit some live preview stuff. This is very irritating. I just don't understand why to enforce it?

It was not intrusive in Brackets and should stay this way in Phoenix, imho :)

**Reply by abose** [reply]
Can you paste a screenshot of the popup?  We have removed that popup in the last release.

**Reply by MerryPanda** [reply]
Sorry, couldn't reproduce. It was at the top of the text area.

**Reply by abose** [reply]
Ah! That is likely custom server previews. If you are editing php or jsp or server rendered files, it comes up once for the project. That was a highly requested feature in the surveys.

**Reply by MerryPanda** [reply]
If I remember correctry it just suggested to turn on the Live Preview. Once again.

It is still unneeded enforcement as you have a special button always available.

This is definitely a bloat.

**Reply by abose** [reply]
Was it a banner or an orange popup? I have removed all the orange popups for.live preview thus release. A screenshot will help more in identifying which ones we are talking about @MerryPanda

**Reply by MerryPanda** [reply]
It was a gray banner at the top of the text area. It still required to click "x" to close it.

Please do not use these irritating pop ups. There is a special button already. Please integrate all functionality into it. It is always available so there are no accesability problems.

**Reply by abose** [reply]
Made change for the next release: It will only be shown when the live preview panel is shown for the first time in a PHP/JSP project. So it should appear a lot less now.

**Comment by ajaxStardust** [comment]
I realize this is an old thread (the struggle decision over whether to post new)
Just wanted to support @MerryPanda 's remark about it potentially being irritating. 
Feedback (and phcode devs probably aware): 
Please get a mastodon acct, as it's fun to interact and communicate things to you about usage, experience with it, etc. I just feel like a fool every time i log into it. Twitter, that is. X if you're hip. 
I see there's an "early features" option list (complete a survey). Just wanted to throw it out there that I'd be willing to do a little multi-media step-by-step setup PHP / live-preview type thing. The whole "Traversy style" video tutorials are new to me, but it's super easy, and practice makes better, so... 
Hit me up!

Best regards! and thank you! it's cooL!
👯

**Reply by abose** [reply]
Its possible via live preview settings. See https://docs.phcode.dev/docs/Features/live-preview-settings/

**Reply by ajaxStardust** [reply]
difficult to say right away why i have issues. 
1.) i dno't use a "normal" setup w/ my LAN -- see "T-Mobile 5G Home", it's nothing more than a hotspot really. not sure if there's a diff. So limited config (not port config i don't believe. just got that, but had been using similar, just my cell hotspot). 
2.) Sometimes using Win WSL, sometimes Debian NGINX via the "LAN" on separate PC. 
So, I don't think the built-in server likes that setup? 

I dunno. i hven't messed w/ it since reading the pge you provided. thank you!

**Reply by ajaxStardust** [reply]
oh! but i did realize-- after making my project small enough to use the "free" (sorry)-- i realize that the online editor seems "better" somehow. at least for me. that's been my experience. been super busy but i want to get back and try that again. i don't recall if it serves the rendered PHP, but i was debugging somethign and fooling around. E.g. the code wouldn't have displayed anyway, unless i'd fixed my bugs. so, bad context there. but, in acny case, it's cool and i'm following!!
I'm a fan!
